js/magnets/MagnetsScreen.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013-2024, University of Colorado Boulder
+
+/**
+ * MagnetsScreen is the top-level component for the 'Magnets' screen.  It creates the model and view.
+ *
+ * @author Chris Malley (PixelZoom, Inc.)
+ */
+
+import Screen, { ScreenOptions } from '../../../joist/js/Screen.js';
+import ScreenIcon from '../../../joist/js/ScreenIcon.js';
+import { Image } from '../../../scenery/js/imports.js';
+import barMagnet_png from '../../images/barMagnet_png.js';
+import ExampleSimColors from '../common/ExampleSimColors.js';
+import ExampleSimConstants, { ExampleSimScreenOptions } from '../common/ExampleSimConstants.js';
+import exampleSim from '../exampleSim.js';
+import ExampleSimStrings from '../ExampleSimStrings.js';
+import MagnetsModel from './model/MagnetsModel.js';
+import MagnetsScreenView from './view/MagnetsScreenView.js';
+import optionize, { EmptySelfOptions } from '../../../phet-core/js/optionize.js';
+
+type SelfOptions = ExampleSimScreenOptions & EmptySelfOptions;
+type MagnetsScreenOptions = SelfOptions & ScreenOptions;
+
+export default class MagnetsScreen extends Screen<MagnetsModel, MagnetsScreenView> {
+
+  public constructor( providedOptions?: MagnetsScreenOptions ) {
+
+    const options = optionize<MagnetsScreenOptions, SelfOptions, ScreenOptions>()(
```

Comment:
`optionize4` is complaining for me. ScreenOptions requires tandem. When I use it like this:

```
    const options = optionize4<MagnetsScreenOptions, SelfOptions, ScreenOptions>()(
      {}, ExampleSimConstants.SCREEN_OPTIONS,
      {
        name: ExampleSimStrings.screen.magnetsStringProperty,
        homeScreenIcon: createScreenIcon()
      },
      providedOptions );
```

It (meaning VS Code with the eslint plugin) says this about the default-values object-literal param:

"Argument of type '{ name: LocalizedStringProperty; homeScreenIcon: ScreenIcon; }' is not assignable to parameter of type 'OptionizeDefaults<EmptySelfOptions, ScreenOptions, EmptySelfOptions>'.
  Property 'tandem' is missing in type '{ name: LocalizedStringProperty; homeScreenIcon: ScreenIcon; }' but required in type 'Required<Omit<Pick<ScreenOptions, "tandem">, never>>'.ts(2345)
PhetioObject.ts(134, 3): 'tandem' is declared here."

Reply:
Ahh great, there are a couple of ways you should be able to address this. You can adjust your code block with either of the following options. I'm going to tag @samreid and @pixelzoom here in case they have any further insight onto how we would like open source contributors to handle tandems.

Option 1:
```
const options = optionize4<MagnetsScreenOptions, SelfOptions, StrictOmit<ScreenOptions, 'tandem'>>()(
      {}, ExampleSimConstants.SCREEN_OPTIONS,
      {
        name: ExampleSimStrings.screen.magnetsStringProperty,
        homeScreenIcon: createScreenIcon()
      },
      providedOptions );
```

Option 2:
```
const options = optionize4<MagnetsScreenOptions, SelfOptions, ScreenOptions>()(
      {}, ExampleSimConstants.SCREEN_OPTIONS,
      {
        name: ExampleSimStrings.screen.magnetsStringProperty,
        homeScreenIcon: createScreenIcon(),
        tandem: Tandem.OPT_OUT
      },
      providedOptions );
```

Reply:
I picked "tandem: Tandem.OPT_OUT" as it seems marginally more clear on what's going on.